Intersect.Server/Maps/MapInstance.cs

```diff
@@ -945,6 +950,40 @@ public void Update(long timeMs)
                     }
 
                     en.Value.Update(timeMs);
+
+                    // Check to see if we need to send any entity vital and status updates for this entity.
+                    if (!en.Value.GetVitals().SequenceEqual(en.Value._oldvital))
+                    {
+                        vitalUpdates.Add(en.Value);
+
+                        // Send a party update if we're a player with a party.
+                        if (en.Value is Player player)
+                        {
+                            for (var i = 0; i < player.Party.Count; i++)
+                            {
+                                PacketSender.SendPartyUpdateTo(player.Party[i], player);
+                            }
+                        }
+                        
+                        en.Value._oldvital = en.Value.GetVitals();
+                    }
+
+                    if (!en.Value.CachedStatuses.SequenceEqual(en.Value.OldCachedStatuses))
```

Comment:
The same comments apply to statuses. This check doesn't really belong in `MapInstance`.

Another idea is

```csharp
statusUpdates.AddRange(en.Value.StatusUpdates)
```
and

```csharp
var entityVitalUpdates= en.Value.VitalUpdates;
if (entityVitalUpdates.Count > 0) {
    // party code
    entityVitalUpdates.AddRange(vitalUpdates);
}
```

Reply:
(And you would do all the checks and updates in those two properties on the entities)